llvm/test/Transforms/IndVarSimplify/2011-10-27-lftrnull.ll

```diff
@@ -50,13 +53,13 @@ entry:
   br label %while.cond
 
 while.cond:
-  br i1 undef, label %while.end, label %while.body
+  br i1 %arg, label %while.end, label %while.body
 
 while.body:                                       ; preds = %while.cond
-  br i1 undef, label %if.then165, label %while.cond
+  br i1 %arg, label %if.then165, label %while.cond
 
 if.then165:                                       ; preds = %while.body
-  br i1 undef, label %while.cond, label %for.body.lr.ph.i81
+  br i1 %arg, label %while.cond, label %for.body.lr.ph.i81
```

Comment:
the icmp undef below makes the br UB

Reply:
@nunoplopes A question here, for this case https://alive2.llvm.org/ce/z/RvwJpB
1. Does infinite loop be seen as Undefined Behavior for Alive2? I remembered John told me that Alive2 unrolls loops twice and after that Alive2 just gives up.

There are two `icmp undef` cases here. The first one was `icmp ult i32 0, undef`, so I replaced it with `false`. The second one was ` icmp ult i32 %inc.i125, undef`, so I replaced it with `%arg`.

Reply:
infinite loops are only UB if the function is marked as terminating (the `mostprogress` attribute).
By default, Alive2 only unrolls once. It can be changed with a flag.

All that said, it would be be best to have loops that are not infinite.